src/invidious/routes/video_playback.cr

```diff
@@ -37,7 +37,8 @@ module Invidious::Routes::VideoPlayback
 
     # See: https://github.com/iv-org/invidious/issues/3302
     range_header = env.request.headers["Range"]?
-    if range_header.nil?
+    sq = query_params["sq"]?
+    if range_header.nil? && sq.nil?
```

Comment:
The `Range` header here is only added for the `HEAD` request below. After that, it's removed again. Does that change really make a difference?

Reply:
Hmm odd, when I tested it without this change before it didn't work but I've just tested it again and now it worked without removing the range header too???

Personally I wouldn't send it, even if it doesn't make a difference, as YouTube's player would never send it for live stream segments either, but it's up to you to decide what the best approach is for Invidious.

Reply:
I should try to get rid of that HEAD request at some point.
I really want to split the logic between DASH and progressive streams/download, but I don't know where to start, this code has always broke on me every time I tried to make significant changes...